src/sql/azdata.proposed.d.ts

```diff
@@ -1255,19 +1258,31 @@ declare module 'azdata' {
 			 */
 			type: DesignerEditType;
 			/**
-			 * the property that was edited
+			 * the path of the edit target.
 			 */
-			property: DesignerEditIdentifier;
+			path: DesignerEditPath;
 			/**
 			 * the new value
 			 */
 			value?: any;
 		}
 
 		/**
-		 * The identifier of a property. The value is string typed if the property belongs to the root object, otherwise the type of the value is an object.
+		 * The path of the edit target.
+		 * Table designer support nested tables, Below are the 3 scenarios and their expected path format.
+		 * Note: 'index-{x}' in the description below represents the index of the object in the list.
+		 * 1. 'Add' scenario
+		 *    a. 'propertyName1'. Example: add a column to the columns property: 'columns'.
+		 *    b. 'propertyName1/index-1/propertyName2'. Example: add a column mapping to the first foreign key: 'foreignKeys/0/mappings'.
+		 * 2. 'Update' scenario
+		 *    a. 'propertyName1'. Example: update the name of the table: 'name'.
+		 *    b. 'propertyName1/index-1/propertyName2'. Example: update the name of a column: 'columns/0/name'.
+		 *    c. 'propertyName1/index-1/propertyName2/index-2/propertyName3'. Example: update the source column of an entry in a foreign key's column mapping table: 'foreignKeys/0/mappings/0/source'.
+		 * 3. 'Remove' scenario
+		 *    a. 'propertyName1/index-1'. Example: remove a column from the columns property: 'columns/0'.
+		 *    b. 'propertyName1/index-1/propertyName2/index-2': Example: remove a column mapping from a foreign key's column mapping table: 'foreignKeys/0/mappings/0'.
 		 */
-		export type DesignerEditIdentifier = string | { parentProperty: string, index: number, property: string };
+		export type DesignerEditPath = string;
```

Comment:
As mentioned in the other PR - any reason you don't just have the type be a string array to begin with? Seems like that'd save a lot of the headache and edge cases for this stuff

Reply:
LOL, this is exactly what I am trying to get out of PRs, I've been trapped in the string solution and need someone to pull me out of it. yes, why not use an array. :)

Reply:
🎉